package/main/daml/Daml.Finance.Interface.Instrument.Base/daml.yaml

```diff
@@ -9,8 +9,8 @@ dependencies:
   - daml-prim
   - daml-stdlib
 data-dependencies:
-  - .lib/daml-finance/Daml.Finance.Interface.Holding/1.99.0.202323.1/daml-finance-interface-holding-1.99.0.202323.1.dar
-  - .lib/daml-finance/Daml.Finance.Interface.Types.Common/1.0.0.99.202323.1/daml-finance-interface-types-common-1.0.0.99.202323.1.dar
+  - .lib/daml-finance/Daml.Finance.Interface.Types.Common/1.0.99.0.202323.1/daml-finance-interface-types-common-1.0.99.0.202323.1.dar
+  - .lib/daml-finance/Daml.Finance.Interface.Types.Date/1.99.0.202323.1/daml-finance-interface-types-date-1.99.0.202323.1.dar
```

Comment:
I think this extra dependency is dangerous. This comes from the fact that
- `CalendarKey` has been moved to `Types.Date`
- the base instrument interface has a new method `getCalendarKeys`

I would suggest the following:
- `CalendarKey` lives in `Interface.Data`
- `getCalendarKeys` is not a method of the base instrument interface, but of the `Claims.Dynamic` interface

This will remove the dependency and will remove a bunch of dummy implementations of `getCalendarKeys` for instruments that have no calendars (Token, Equity, Generic)

Reply:
wanted to be able to use the same (script) function to build the index table, but got it working using:
```
case fromInterface @DynamicInstrument.I i of
    None -> pure idx
    Some dynamicInstrument -> ...  -- adding calendar keys to the index table
```
reading from the documentation, it was not clear that it also works for trying to convert to an interface value.